### PR TITLE
[MPS] Speedup addmm

### PR DESCRIPTION
**Pull request by malfet**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #116548
* #116547

- Do not copy bias to output
- Skip respective multiplication op if either alpha or beta are equal to 1.0

### CODE REVIEW

File: aten/src/ATen/native/mps/operations/LinearAlgebra.mm [hunk 1]
```diff
@@ -247,9 +247,6 @@
 
   if (&output != &self) {
     output.resize_(bias_sizes);
-    if (beta.toComplexDouble() != 0.0) {
-      output.copy_(*bias_);
```
Comment (albanD) [inline aten/src/ATen/native/mps/operations/LinearAlgebra.mm:251]: If this is an out variant, overriding the output completely is not ok. We should add into it.

Comment (malfet) [reply aten/src/ATen/native/mps/operations/LinearAlgebra.mm:251]: Yes, and it  is(all `torch.addmm` OpInfo tests use `alpha` and `beta` not equal to 1), this is why I'm removing this one, as copy is redundant.

Comment (albanD) [reply aten/src/ATen/native/mps/operations/LinearAlgebra.mm:251]: So I guess the next steps here would be to add this case to OpInfo and fix this kernel to have the appropriate behavior?

Comment (malfet) [reply aten/src/ATen/native/mps/operations/LinearAlgebra.mm:251]: I don't think anything is needed here: `addmm_out_mps` copied bias to output for some reason, even though it wasn't at all needed, as MPSGraph always overwrites the output;
https://github.com/pytorch/pytorch/blob/035e55822ad123b02fbd9d91e1185f5c07af0ddd/aten/src/ATen/native/mps/operations/LinearAlgebra.mm#L302-L305

Comment (malfet) [reply aten/src/ATen/native/mps/operations/LinearAlgebra.mm:251]: Hmm, or perhaps I did not understand your original question: this override only happens if output != self, though yes, I'm not sure this kernel will work as expected if output = self, but imo this should be done as separate PR, this just eliminates unneeded multiplications if alpha and beta are 1 and unneeded override as function always writes to output

**Review by albanD — APPROVED** [review]
Sounds good!

**Pull request closed without merging**